File: jdroid-gradle-googleplay-publisher-plugin/src/main/java/com/jdroid/android/googleplay/publisher/task/ListTracksTask.java
package com.jdroid.android.googleplay.publisher.task;

import com.google.api.services.androidpublisher.model.Track;
import com.google.api.services.androidpublisher.model.TrackRelease;
import com.jdroid.android.googleplay.publisher.App;
import com.jdroid.android.googleplay.publisher.GooglePlayPublisher;

import org.gradle.api.logging.LogLevel;

public class ListTracksTask extends BaseTask {
	
	private LogLevel logLevel = LogLevel.LIFECYCLE;
	
	public ListTracksTask() {
		setDescription("List all the tracks and its releases");
	}

	@Override
	protected void onExecute(App app) {
		for (Track track : GooglePlayPublisher.getTracks(app)) {
			if (track.getReleases() != null) {
				getLogger().log(logLevel, "Track: " + track.getTrack());
				getLogger().log(logLevel, "-------------------------------");
				for (TrackRelease trackRelease : track.getReleases()) {
					getLogger().log(logLevel, "  Release name: " + trackRelease.getName());
					getLogger().log(logLevel, "  Version codes: " + trackRelease.getVersionCodes());
					getLogger().log(logLevel, "  User fraction: " + trackRelease.getUserFraction());
					getLogger().log(logLevel, "  Status: " + trackRelease.getStatus());
					getLogger().log(logLevel, "  Release Notes: " + trackRelease.getReleaseNotes());
					getLogger().log(logLevel, "");
				}
				getLogger().log(logLevel, "");
			}
		}
	}

}

File: jdroid-gradle-googleplay-publisher-plugin/src/main/java/com/jdroid/android/googleplay/publisher/task/ListTracksTask.kt
package com.jdroid.android.googleplay.publisher.task

import com.jdroid.android.googleplay.publisher.App
import com.jdroid.android.googleplay.publisher.GooglePlayPublisher
import org.gradle.api.logging.LogLevel

open class ListTracksTask : BaseTask() {

    private val logLevel = LogLevel.LIFECYCLE

    init {
        description = "List all the tracks and its releases"
    }

    override fun onExecute(app: App) {

        GooglePlayPublisher.getTracks(app).forEach { track ->
            if (track.releases != null) {
                logger.log(logLevel, "Track: " + track.track)
                logger.log(logLevel, "-------------------------------")
                track.releases.forEach { trackRelease ->
                    logger.log(logLevel, "  Release name: " + trackRelease.name)
                    logger.log(logLevel, "  Version codes: " + trackRelease.versionCodes)
                    logger.log(logLevel, "  User fraction: " + trackRelease.userFraction)
                    logger.log(logLevel, "  Status: " + trackRelease.status)
                    logger.log(logLevel, "  Release Notes: " + trackRelease.releaseNotes)
                    logger.log(logLevel, "")
                }
                logger.log(logLevel, "")
            }
        }
    }
}
